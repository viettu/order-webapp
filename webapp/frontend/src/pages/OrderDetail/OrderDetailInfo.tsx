import { TableContainer, Table, Tr, Td, Tbody, Box } from '@chakra-ui/react';
import { IOrderInfo } from '../../data';

type OrderDetailInfoProps = {
  orderInfo: IOrderInfo;
};

const OrderDetailInfo: React.FC<OrderDetailInfoProps> = ({ orderInfo }) => {
  if (!orderInfo) {
    return null;
  }

  return (
    <Box border={'1px solid lightgray'} borderRadius={'20px'} p={2}>
      <TableContainer>
        <Table variant="unstyled" colorScheme="teal">
          <Tbody>
            <Tr>
              <Td p={2} width={'100px'} fontWeight={'bold'}>
                Name:
              </Td>
              <Td p={2}>{orderInfo.name}</Td>
            </Tr>
            <Tr>
              <Td p={2} fontWeight={'bold'}>
                Phone:
              </Td>
              <Td p={2}>{orderInfo.phone}</Td>
            </Tr>
            <Tr>
              <Td p={2} fontWeight={'bold'}>
                Address:
              </Td>
              <Td p={2}>{orderInfo.address}</Td>
            </Tr>
          </Tbody>
        </Table>
      </TableContainer>
    </Box>
  );
};

export default OrderDetailInfo;
