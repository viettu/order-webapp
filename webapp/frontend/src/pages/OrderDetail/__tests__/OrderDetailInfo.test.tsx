import React from 'react';
import OrderDetailInfo from '../OrderDetailInfo';
import renderer from 'react-test-renderer';
import { TestComponentWrapper } from '../../../utils';

test('snapshot', () => {
  const orderInfo = {
    name: 'Viet Tu',
    address: 'Address',
    phone: '09090909',
  };
  const component = renderer.create(
    <TestComponentWrapper>
      <OrderDetailInfo orderInfo={orderInfo}></OrderDetailInfo>
    </TestComponentWrapper>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
