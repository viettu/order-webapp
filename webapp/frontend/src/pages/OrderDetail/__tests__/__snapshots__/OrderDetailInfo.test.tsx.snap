// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Array [
  <div
    className="css-ydi3wo"
  >
    <div
      className="chakra-table__container css-zipzvv"
    >
      <table
        className="chakra-table css-5605sr"
        role="table"
      >
        <tbody
          className="css-0"
        >
          <tr
            className="css-0"
            role="row"
          >
            <td
              className="css-pb2ry3"
              role="gridcell"
            >
              Name:
            </td>
            <td
              className="css-1hwcto5"
              role="gridcell"
            >
              Viet Tu
            </td>
          </tr>
          <tr
            className="css-0"
            role="row"
          >
            <td
              className="css-p9na44"
              role="gridcell"
            >
              Phone:
            </td>
            <td
              className="css-1hwcto5"
              role="gridcell"
            >
              09090909
            </td>
          </tr>
          <tr
            className="css-0"
            role="row"
          >
            <td
              className="css-p9na44"
              role="gridcell"
            >
              Address:
            </td>
            <td
              className="css-1hwcto5"
              role="gridcell"
            >
              Address
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  <span />,
  <span
    className="chakra-env"
    hidden={true}
  />,
]
`;
